instrumentation/redis/lib/opentelemetry/instrumentation/redis/utils.rb

```diff
@@ -17,24 +17,46 @@ module Utils
 
         def format_span_name(commands)
           commands.map do |command|
-            format_arg(resolve_command_args(command).first)
+            format_cmd(resolve_command_args(command).first)
           end.join(' ')
         end
 
         def format_statement(command_args)
           command_args = resolve_command_args(command_args)
           return 'AUTH ?' if auth_command?(command_args)
 
-          cmd = command_args.map { |x| format_arg(x) }.join(' ')
-          OpenTelemetry::Common::Utilities.truncate(cmd, CMD_MAX_LEN)
+          cmd = format_cmd(command_args.first)
+          args = command_args[1..-1].map { |arg| format_arg(arg) }
+          statement = args.unshift(cmd).join(' ')
+          OpenTelemetry::Common::Utilities.truncate(statement, CMD_MAX_LEN)
+        end
+
+        def format_cmd(cmd)
+          format_string do
+            cmd.to_s.upcase
+          end
         end
 
         def format_statements(commands)
           commands.map { |command| format_statement(command) }.join("\n")
         end
 
         def format_arg(arg)
-          str = arg.is_a?(Symbol) ? arg.to_s.upcase : arg.to_s
+          format_string do
+            obfuscate_arg(arg.to_s)
+          end
+        end
+
+        private
+
+        def obfuscate_arg(arg)
+          return PLACEHOLDER if config[:enable_statement_obfuscation]
+
+          arg
+        end
+
+        def format_string(input = nil)
+          str = block_given? ? yield(input) : input
```

Comment:
This seems over engineered - I don't think this should accept a block. Both call sites can pass an immediate arg.

Reply:
Part of my original intent with this was to ensure that exceptions in converting the strings above would be caught here. However, I'm now thinking my memory of to_s might be a little shaky and pretty much everything can do it.

Reply:
It looks like swapping this caused an internal test to error. You can see in the GH Actions output.

From what I understand `to_s` should be defined on every object except for `BasicObject` so this failure is very unlikely to occur. Are you happy for me to remove the test?